plugins/inputs/irqstat/irqstat.go

```diff
@@ -0,0 +1,126 @@
+package irqstat
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"io/ioutil"
+	"strconv"
+	"strings"
+)
+
+type Irqstat struct {
+	Include []string
+}
+
+type IRQ struct {
+	ID     string
+	Type   string
+	Device string
+	Values map[string]interface{}
+}
+
+func NewIRQ(id string) *IRQ {
+	return &IRQ{ID: id, Values: make(map[string]interface{})}
+}
+
+const sampleConfig = `
+  ## A list of IRQs to include for metric ingestion, if not specified
+  ## will default to collecting all IRQs.
+  # include = ["0", "1", "30", "NET_RX"]
+`
+
+func (s *Irqstat) Description() string {
+	return "This plugin gathers interrupts data from /proc/interrupts and /proc/softirqs."
+}
+
+func (s *Irqstat) SampleConfig() string {
+	return sampleConfig
+}
+
+func parseInterrupts(irqdata string, include []string) ([]IRQ, error) {
+	var err error
+	var irqs []IRQ
+	var cpucount int
+	scanner := bufio.NewScanner(strings.NewReader(irqdata))
+	for scanner.Scan() {
+		var irqval, irqtotal int64
+		var irqtype, irqdevice string
+		fields := strings.Fields(scanner.Text())
+		if len(fields) > 0 && fields[0] == "CPU0" {
+			cpucount = len(fields)
+		}
+		if !strings.HasSuffix(fields[0], ":") {
+			continue
+		}
+		irqid := strings.TrimRight(fields[0], ":")
+		irq := NewIRQ(irqid)
+		_, err := strconv.ParseInt(irqid, 10, 64)
+		if err == nil && len(fields) > cpucount {
+			irqtype = fields[cpucount+1]
+			irqdevice = strings.Join(fields[cpucount+2:], " ")
+		} else if len(fields) > cpucount {
+			irqtype = strings.Join(fields[cpucount+1:], " ")
+		}
+		fields = fields[1:len(fields)]
+		for i := 0; i < cpucount; i++ {
+			cpu := fmt.Sprintf("CPU%d", i)
+			if i < len(fields) {
+				irqval, err = strconv.ParseInt(fields[i], 10, 64)
+				if err != nil {
+					return irqs, err
```

Comment:
I think we should add context to the error, so that we know what failed to parse.
Maybe something like `fmt.Errorf("unable to parse %q from %q: %s", fields[i], scanner.Text(), err)`.